packages/builtin/array.pony

```diff
@@ -63,11 +63,11 @@ class Array[A] is Seq[A]
     """
     _ptr._offset(from_offset)._copy_to(ptr._offset(to_offset), copy_len)
 
-  fun cpointer(): Pointer[A] tag =>
+  fun cpointer(offset: USize = 0): Pointer[A] tag =>
     """
     Return the underlying C-style pointer.
     """
-    _ptr
+    _ptr._offset(offset)
```

Comment:
I'm just thinking it through, and I believe there's nothing you can do with a `Pointer tag` that would result in (read or write) access to memory, without using FFI.  So if you end up with a bogus `Pointer` to memory that would lead to a segfault if accessed, it wouldn't actually cause a segfault until you tried to access memory with it (which can't be done without FFI), right?